### PR TITLE
Add Participant Project Species to Search API

### PR DESCRIPTION
**Pull request by nickgraz**
- Add `participantProjectSpecies` to both `projects` and `species` as a sublist

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#1937** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1937?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1935** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1935?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1934** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1934?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1933** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1932** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1932?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1924** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1924?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1919** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1919?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1918** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1918?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1917** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1917?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1916** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1916?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by nickgraz** [comment]
@sgrimm I have been having trouble to make this work. There seems to be something I am doing that is causing it to run out of memory. Locally, the errors look like several random concurrently running tests exit. 

I have tried changing the `get()` call to use an "embeddable definition type" with this configuration:

```
EmbeddableDefinitionType()
    .withName("participant_project_species_id")
    .withTables("accelerator.participant_project_species")
    .withColumns("id","project_id", "species_id"),
```

I also tried a variation of that where the columns only included `project_id` and `species_id` but that did not fix it either.

Is there something obvious that I am doing wrong here? 

My intent is to add the capability to access projects associated to species via `prefix: "species", field: "participantProjectSpecies.projects.name"` and species associated to projects via `prefix: "projects", field: "participantProjectSpecies.species.scientificName".

File: src/main/kotlin/com/terraformation/backend/search/table/ParticipantProjectSpeciesTable.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  override val primaryKey: TableField<out Record, out Any?>
+    get() = PARTICIPANT_PROJECT_SPECIES.ID
+
+  override val sublists: List<SublistField> by lazy {
+    with(tables) {
+      listOf(
+          projects.asMultiValueSublist(
+              "projects", PARTICIPANT_PROJECT_SPECIES.PROJECT_ID.eq(PROJECTS.ID)),
+          species.asMultiValueSublist(
+              "species", PARTICIPANT_PROJECT_SPECIES.SPECIES_ID.eq(SPECIES.ID)),
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/search/table/ParticipantProjectSpeciesTable.kt]: These should both be single-value sublists. There can only be one project and one species for a given participant project species.

Changing that ought to fix the failing test.

**Review by sgrimm — APPROVED** [review]

**Comment by nickgraz** [comment]
### Merge activity

* **May 8, 12:09 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1937).
* **May 8, 12:43 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1937) rebased this pull request as part of a merge.
* **May 8, 12:52 PM EDT**: @nickgraz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1937).

**Pull request merged**